src/create-glamorous.js

```diff
@@ -81,6 +87,30 @@ function createGlamorous(splitProps) {
         })(GlamorousComponent.styles)
       }
 
+      function shouldUpdate(props, context) {
```

Comment:
This looks like a clever solution, but something about it rubs me the wrong way. It very well may be that I just haven't seen it before. I would suggest calling it something other than state though. I can see this getting confusing when developing and debugging.